pkg/controller/changetierrequest/changetierrequest_controller.go

```diff
@@ -74,5 +81,122 @@ func (r *ReconcileChangeTierRequest) Reconcile(request reconcile.Request) (recon
 		return reconcile.Result{}, err
 	}
 
-	return reconcile.Result{}, nil
+	// if is complete, then we can delete it
+	if condition.IsTrue(changeTierRequest.Status.Conditions, toolchainv1alpha1.ChangeTierRequestComplete) {
```

Comment:
We will never get to that, will we? Our predicate will trigger reconcile only for NotComplete resources.

Reply:
> Our predicate will trigger reconcile only for NotComplete resources.

I'm not 100% sure how it will behave if:
1. the tier has been changed
2. the status is set to complete
3. the deletion of the ChangeTierRequest fails
4. we return the error, but keeping the status = complete
5. will be the reconcile triggered again based on the error or filtered out based on the predicate?

This is the reason for this check - just to be on the safe side :-)